Source/Modules/php.cxx

```diff
@@ -441,6 +457,9 @@ class PHP : public Language {
     Printf(s_header, "}\n");
     Printf(s_header, "#endif\n\n");
 
+    Printf(s_header,"#ifdef __GNUG__\n#define SWIG_remove(zv) delete zv\n");
+    Printf(s_header,"#else\n#define SWIG_remove(zv) free(zv)\n#endif\n\n");
```

Comment:
The correct test for "is this a C++ compiler?" is `#ifdef __cplusplus`.

You are instead testing: "is this GNU C++?"